src/objects/zcl_abapgit_object_iobj.clas.abap

```diff
@@ -15,26 +15,46 @@ CLASS zcl_abapgit_object_iobj IMPLEMENTATION.
 
   METHOD zif_abapgit_object~changed_by.
 
-    DATA: ls_viobj TYPE rsd_s_viobj,
-          lv_objna TYPE rsd_iobjnm.
+    DATA: lv_objna TYPE c LENGTH 30,
+          lr_viobj TYPE REF TO data,
+          lv_type  TYPE string.
 
+    FIELD-SYMBOLS:
+      <lv_tstpnm> TYPE any,
+      <ls_viobj>  TYPE any.
+
+    lv_type  = 'rsd_s_viobj'.
     lv_objna = ms_item-obj_name.
 
+    TRY.
+        CREATE DATA lr_viobj TYPE (lv_type).
```

Comment:
the variable lv_type can be removed by doing

```suggestion
        CREATE DATA lr_viobj TYPE ('RSD_S_VI_OBJ').
```

Reply:
Done